pkg/minikube/config/profile.go

```diff
@@ -35,22 +36,31 @@ func (p *Profile) IsValid() bool {
 	if p.Config == nil {
 		return false
 	}
-	if len(p.Config) == 0 {
-		return false
-	}
-	// This will become a loop for multinode
-	if p.Config[0] == nil {
+	if p.Config == nil {
 		return false
 	}
-	if p.Config[0].VMDriver == "" {
+	if p.Config.VMDriver == "" {
 		return false
 	}
-	if p.Config[0].KubernetesConfig.KubernetesVersion == "" {
-		return false
+	for _, n := range p.Config.Nodes {
+		if n.KubernetesVersion == "" {
+			return false
+		}
 	}
 	return true
 }
 
+// MasterNode gets the node specific config for the control plane
```

Comment:
mention the behavior if there is more than one control-plane machine. I understand we do not support this configuration yet, however.

Instead of `master`, how about `Primary` for this case?